scripts/jenkins/groovy/defineTestStages.groovy

```diff
@@ -311,6 +311,10 @@ def call(final pipelineContext) {
     [
       stageName: 'R3.3 CMD Check as CRAN', target: 'test-r-cmd-check-as-cran', rVersion: '3.3.3',
       timeoutValue: 10, hasJUnit: false, component: pipelineContext.getBuildConfig().COMPONENT_R
+    ],
+    [ // These run with reduced number of file descriptors for early detection of FD leaks
+      stageName: 'XGBoost Stress tests', target: 'test-pyunit-xgboost-stress', pythonVersion: '3.5', timeoutValue: 20,
+      component: pipelineContext.getBuildConfig().COMPONENT_PY, customDockerArgs: [ '--ulimit nofile=100:100' ]
```

Comment:
Notice the `'--ulimit nofile=100:100'`. Makes this test fail fast if there is something wrong with resources not being freed. If these tests are ran without the fixes, they fail in less than 100 models built (as some of the descriptors are already taken by the system).